ipaserver/install/ipa_migrate_constants.py

```diff
@@ -0,0 +1,938 @@
+            'cn=certificate store issuer/serial uniqueness,cn=plugins,'
+            'cn=config',
+            'cn=caacl name uniqueness,cn=plugins,cn=config',
+            'cn=netgroup uniqueness,cn=plugins,cn=config',
+            'cn=sudorule name uniqueness,cn=plugins,cn=config',
+            'cn=ipaSubordinateIdEntry ipaOwner uniqueness,cn=plugins,'
+            'cn=config',
+            'cn=mail uniqueness,cn=plugins,cn=config',
+        ],
+        'attrs': [
+            'uniqueness-across-all-subtrees',
+        ],
+        'multivalued': [
+            'uniqueness-subtrees', 'uniqueness-exclude-subtrees',
+            'uniqueness-attribute-name'
+        ],
+        'label': 'Attr Uniqueness Plugin',
+        'mode': 'all',
+        'count': 0,
+    },
+}
+
+#
+# Slpai NIS is an optional plugin.  It requires special handling
+#
+NIS_PLUGIN = {
+    'dn': 'cn=NIS Server,cn=plugins,cn=config',
+    'attrs': [
+        'nis-domain',
+        'nis-base',
+        'nis-map',
+        'nis-filter',
+        'nis-key-format:',
+        'nis-values-format:',
+        'nis-secure',
+        'nis-disallowed-chars',
+        # Parent plugin entry
+        'nsslapd-pluginarg0',
+        'nsslapd-pluginenabled'
+    ],
+    'multivalued': [],
+    'label': 'NIS Server Plugin',
+    'mode': 'all',
+    'count': 0,
+}
+
+#
+# This mapping is simliar to above but it handles container entries
+# This could be built into the above mapping using the "comma" approach
+#
+DS_INDEXES = {
+    'index': {
+        'dn': ',cn=index,cn=userroot,cn=ldbm database,cn=plugins,cn=config',
+        'attrs': [
+            'nssystemindex',
+        ],
+        'multivalued': [
+            'nsindextype',
+            'nsmatchingrule',
+        ],
+        'label': 'Database Indexes',
+        'mode': 'all',
+        'count': 0
+    },
+    'encrypted': {
+        'dn': ',cn=encrypted attributes,cn=userroot,cn=ldbm database,'
+              'cn=plugins,cn=config',
+        'attrs': [
+            'nsencryptionalgorithm',
+        ],
+        'multivalued': [],
+        'label': 'Encrypted Attributes',
+        'mode': 'all',
+        'count': 0
+    },
+}
+
+#
+# This mapping breaks each IPA entry (or type of entry) into its own catagory
+# Each catagory, or type, has an objectclass list "oc" and its DIT location
+# "subtree".  If the "subtree" starts with a comma "," then it is a container
+# of entries, otherwise it's a single entry.  These two are used together to
+# identify the entry.
+# The "label" and "count" attributes are used for the Summary Report
+#
+DB_OBJECTS = {
+    # Plugins
+    'automember': {
+        'oc': ['automemberdefinition'],
+        'subtree': ',cn=automember,cn=etc,$SUFFIX',
+        'label': 'Automember Definitions',
+        'mode': 'all',
+        'count': 0,
+    },
+    'dna_ranges': {
+        'oc': ['ipadomainidrange', 'ipaidrange', 'ipatrustedaddomainrange'],
+        'subtree': ',cn=ranges,cn=etc,$SUFFIX',
```

Comment:
We try not to use strings as DNs. There is a class to handle this, ipapython.dn.DN. The subtree values are all in ipalib/constants.py (or should be). You can re-use those.

Reply:
> We try not to use strings as DNs. There is a class to handle this, ipapython.dn.DN. The subtree values are all in ipalib/constants.py (or should be). You can re-use those.

Well if the "subtree" starts with a comma then its looking for child entries, otherwise it is only looking for that exact DN.  I'm not sure the DN class would work this way.